webui/src/pages/repositories/repository/fileRenderers/data.tsx

```diff
@@ -26,24 +28,52 @@ LIMIT 20`
 FROM read_csv(lakefs_object('${repoId}', '${refId}', '${path}'), DELIM='\t', AUTO_DETECT=TRUE) 
 LIMIT 20`
     }
-    const [query, setQuery] = useState<string>(initialQuery)
     const [shouldSubmit, setShouldSubmit] = useState<boolean>(true)
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const [data, setData] = useState<any[] | null>(null);
     const [error, setError] = useState<string | null>(null)
+    const [loading, setLoading] = useState<boolean>(false)
+
+    const sql = useRef<HTMLTextAreaElement>(null);
+
+    const handleSubmit = useCallback((event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        setShouldSubmit(!shouldSubmit)
+    }, [setShouldSubmit])
 
     useEffect(() => {
+        setLoading(true)
         withConnection(async conn => {
-            const results = await conn.query(query)
+            if (!sql || !sql.current) return
+            const results = await conn.query(sql.current.value)
             const data = results.toArray()
             setData(data)
             setError(null)
+
         }).catch(e => {
             setError(e.toString())
+        }).finally(() => {
+            setLoading(false)
         })
     }, [repoId, refId, path, shouldSubmit])
 
+
     let content;
+    let button = (
```

Comment:
IMO something along these lines would be cleaner and more readable.
```suggestion
    const button = (
      <Button type="submit" variant="success" disabled={loading}>
        <DatabaseIcon /> {" "}
        { loading ? "Executing..." : "Execute" }
      </Button>
    );
```

Reply:
Nice! used.